experiments/customui/parent.js

```diff
@@ -250,30 +250,78 @@ var ex_customui = class extends ExtensionCommon.ExtensionAPI {
       return result;
     };
 
-    // Enables dynamic height and fixed 100% width for a WebExtension frame
-    const setWebextFrameSizesForVerticalBox = function(frame, options) {
-      frame.width = "100%";
-      frame.height = (options.height || 100) + "px";
-      frame.style.display = options.hidden ? "none" : "block";
+    // Set a dimension of the frame given in options
+    //  and enable changing by CustomUILocalOptionsListener
+    // If dimension it is not provided in options (a number from API call, in pixels),
+    //  set default value: defaultValue (must include the unit)
+    // dimensionName: "width" or "height"
+    // Only to be called by setWebextFrameDimension
+    const setWebextFrameDynamicDimension = function(frame, options, dimensionName, defaultValue) {
+      frame[dimensionName] = options[dimensionName]
+          ? options[dimensionName] + "px" : defaultValue;
       frame.addCustomUILocalOptionsListener(lOptions => {
-        if (typeof lOptions.height === "number") {
-          frame.height = lOptions.height + "px";
-          frame.style.height = frame.height;
+        if (typeof lOptions[dimensionName] === "number") {
+          frame[dimensionName] = lOptions[dimensionName] + "px";
+          frame.style[dimensionName] = frame[dimensionName];
         }
         if (typeof lOptions.hidden === "boolean") {
```

Comment:
I missed this in my earlier reviews, but handling `"hidden"` should not depend on having a dynamic dimension (nor get doubled when a frame is dynamic in both directions). (I will address this in a proposal commit.)

Reply:
This bothered me too while I was writing the TODO with the suggestion of creating 1 function for the listener and the direct setting. You're doing a great job on this PR! :)